text/0122-sdk-hub-scope-merge.md

```diff
@@ -0,0 +1,475 @@
+
+From a users perspective, this should mostly not be noticeable - they can always run `getCurrentScope()` to get the current scope, or `withScope(callback)` to fork a new scope off the current scope.
+
+You can make a scope the current one via `setCurrentScope(scope)`, which should bind the scope to the current execution context (or a global, in SDKs without execution context). This is a replacement for the current APIs like `makeMain(hub)` or `setCurrentHub(hub)`.
+
+You can still clone scopes manually the same way as before, e.g. via `Scope.clone(oldScope)` or a similar API. In contrast to `withScope()`, this will _not_ fork an execution context.
+
+You can update the client of a scope via `scope.setClient(newClient)`. This will not affect any scope that has already been forked off this scope, but any scope forked off _after_ the client was updated will also receive the updated client.
+
+A current scope always inherits from the previous current scope. 
+
+## Special Scopes
+
+In addition to the current scope, this RFC introduces two new concepts:
+
+- [Global Scope](#global-scope)
+- [Isolation Scope](#isolation-scope)
+
+These scopes will be explained in detail below. They are _additional concepts_ that can be used by SDK internals & power users to make sure data can be assigned to the correct place, without relying on internals about how/when/where a scope or execution context is forked.
+
+The following chart gives an overview of how these scopes work and to which events they would be applied:
+
+![image](./0122-images/scope-inheritance.png)
+
+### How to store/reference scopes
+
+Similar to how we currently store the current hub, we also need to store the current scope. The exact mechanism to do that is up to the concrete SDK/language. In environments with multiple execution contexts (or thread locals), we should attach the current scope to the execution context (or thread local). In other environments, it may be attached to some global (e.g. in Browser JS). This RFC does not propose to change this, so generally the current scope may be stored the same was as the current hub used to be.
+
+In addition to the current scope, the current [Isolation Scope](#isolation-scope) (see below for an in-depth explanation on what these are) also needs to be stored. This should be stored _next to_ the current scope. If you have executution context, you may store/access them like this:
+
+```js
+const executionContext1 = getExecutionContext();
+const { scope, isolationScope } = executionContext1.getScopes();
+executionContext1.setScopes({ scope, isolationScope });
+```
+
+In environments where you don't have execution contexts, you may store them in a global place. The exact mechanism to do this is up to the concrete SDK/language.
+
+For each execution context, there will _always_ be exactly one current & isolation scope. When the execution context is forked, these are forked along.
```

Comment:
I feel like this can easily be misinterpreted as if you fork execution context (e.g. because a new span was started) you always have to fork isolation scope and current scope.

Maybe we should add a chapter about forking scopes as well as passing scopes along when e.g. changing to another thread. I feel like whenever we fork an isolation scope we should also fork the current scope as otherwise we'd open the door for scope bleed and always require users to call both `withIsolationScope` and `withScope` in combination. When moving to another thread we should bring along both isolation and current scope to avoid scope bleed from things that happened on that thread before.